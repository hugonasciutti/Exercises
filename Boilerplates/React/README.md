# Features

Dry boiler plate with:

- Webpack dev server
- React
- Babel Env
