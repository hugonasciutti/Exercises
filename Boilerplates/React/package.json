{
  "name": "boilerplatereact",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "webpack-dev-server"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "final-form": "^4.5.2",
    "react": "^16.3.2",
    "react-dom": "^16.3.2"
  },
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.4",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "html-webpack-plugin": "^3.2.0",
    "webpack": "^4.6.0",
    "webpack-cli": "^2.0.15",
    "webpack-dev-server": "^3.1.3"
  }
}
