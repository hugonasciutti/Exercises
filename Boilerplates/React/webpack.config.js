const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  mode: 'development',

  entry: './src/index.jsx',

  output: {
    path: `${__dirname}/dist`,
    filename: 'app.js'
  },

  devtool: 'source-map',

  module: {
    rules: [
      { test: /\.(js|jsx)?$/, loader: 'babel-loader', exclude: /node_modules/ }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      title: 'Boilerplate',
      template: './src/index.ejs'
    })
  ]
}
