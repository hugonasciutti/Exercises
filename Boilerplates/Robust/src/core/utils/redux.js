export const createReducer = (initialState, actions) =>
  (state = initialState, action) => {
    const reducer = actions[action.type]
    return reducer ? reducer(state, action) : state
  }

export default createReducer
