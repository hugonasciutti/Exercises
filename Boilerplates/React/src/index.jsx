import React, { Component } from 'react'
import { render } from 'react-dom'
import Form from './Form'

class App extends Component {
  render() {
    return (
      <div>
        <Input />
      </div>
    )
  }
}

render(<App />, document.getElementById('app'))