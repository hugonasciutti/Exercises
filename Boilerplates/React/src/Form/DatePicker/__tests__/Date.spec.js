import React from 'react'
import { shallow } from 'enzyme'
import moment from 'core/utils/moment'
import { buildEvent } from 'core/unit-test/utils'
import '../../../../unit-test/fixture/datepicker'
import { Date } from '../'

describe.skip('Date Component', () => {
  it('should render', () => {
    const isoDate = moment().toISOString()
    const component = shallow(<Date value={isoDate} onChange={jest.fn()} />)

    expect(component).toMatchSnapshot()
  })

  it('should render and replace date format', () => {
    const isoDate = moment().toISOString()
    const component = shallow(<Date value={isoDate} onChange={jest.fn()} dateFormat="MM/DD/Y" />)

    expect(component).toMatchSnapshot()
  })

  it('should render with placeholder', () => {
    const isoDate = moment().toISOString()
    const component = shallow(<Date value={isoDate} onChange={jest.fn()} placeholder="Date" />)

    expect(component).toMatchSnapshot()
  })

  it('should change date', () => {
    const date = moment()
    const onChange = jest.fn()
    const component = shallow(<Date value={undefined} onChange={onChange} />)

    component.simulate('change', date)

    expect(onChange).toHaveBeenCalledWith(date.toISOString())
  })

  it('should change empty date', () => {
    const onChange = jest.fn()
    const component = shallow(<Date value={undefined} onChange={onChange} />)

    component.simulate('change', '')

    expect(onChange).toHaveBeenCalledWith('')
  })

  it('should blur date', () => {
    const invalidDate = '16/03/199222'
    const onChange = jest.fn()
    const component = shallow(<Date value={undefined} onChange={onChange} />)

    component.simulate('blur', buildEvent(invalidDate))

    expect(onChange).toHaveBeenCalledWith('')
  })
})
