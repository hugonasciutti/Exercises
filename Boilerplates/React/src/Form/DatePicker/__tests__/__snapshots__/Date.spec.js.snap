// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Component should render 1`] = `
<DatePicker
  className="rw-date form-control"
  dateFormat="MM/DD/YYYY"
  fixedHeight={true}
  onBlur={[Function]}
  onChange={[Function]}
  required={false}
  selected={"1970-01-01T00:00:00.000Z"}
/>
`;

exports[`Date Component should render and replace date format 1`] = `
<DatePicker
  className="rw-date form-control"
  dateFormat="MM/DD/Y"
  fixedHeight={true}
  onBlur={[Function]}
  onChange={[Function]}
  required={false}
  selected={"1970-01-01T00:00:00.000Z"}
/>
`;

exports[`Date Component should render with placeholder 1`] = `
<DatePicker
  className="rw-date form-control"
  dateFormat="MM/DD/YYYY"
  fixedHeight={true}
  onBlur={[Function]}
  onChange={[Function]}
  placeholderText="Date"
  required={false}
  selected={"1970-01-01T00:00:00.000Z"}
/>
`;
